src/hyp3lib/dem.py

```diff
@@ -0,0 +1,227 @@
+import json
+from collections.abc import Generator
+from pathlib import Path
+from subprocess import PIPE, run
+from tempfile import TemporaryDirectory
+
+from osgeo import gdal, ogr, osr
+
+from hyp3lib import DemError
+
+
+DEM_GEOJSON = '/vsicurl/https://asf-dem-west.s3.amazonaws.com/v2/cop30-2021-with-cop90-us-west-2-mirror.geojson'
+GEOID = '/vsicurl/https://asf-dem-west.s3.amazonaws.com/GEOID/us_nga_egm2008_1.tif'
+
+gdal.UseExceptions()
+ogr.UseExceptions()
+
+
+class GDALConfigManager:
+    """Context manager for setting GDAL config options temporarily"""
+
+    def __init__(self, **options):
+        """Args:
+        **options: GDAL Config `option=value` keyword arguments.
+        """
+        self.options = options.copy()
+        self._previous_options = {}
+
+    def __enter__(self):
+        for key in self.options:
+            self._previous_options[key] = gdal.GetConfigOption(key)
+
+        for key, value in self.options.items():
+            gdal.SetConfigOption(key, value)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for key, value in self._previous_options.items():
+            gdal.SetConfigOption(key, value)
+
+
+def get_geometry_from_kml(kml_file: str) -> ogr.Geometry:
+    cmd = [
+        'ogr2ogr',
+        '-wrapdateline',
+        '-datelineoffset',
+        '20',
+        '-f',
+        'GeoJSON',
+        '-mapfieldtype',
+        'DateTime=String',
+        '/vsistdout',
+        kml_file,
+    ]
+    geojson_str = run(cmd, stdout=PIPE, check=True).stdout
+    geometry = json.loads(geojson_str)['features'][0]['geometry']
+    return ogr.CreateGeometryFromJson(json.dumps(geometry))
+
+
+def get_dem_features() -> Generator[ogr.Feature, None, None]:
+    ds = ogr.Open(DEM_GEOJSON)
+    layer = ds.GetLayer()
+    for feature in layer:
+        yield feature
+    del ds
+
+
+def intersects_dem(geometry: ogr.Geometry) -> bool:
+    for feature in get_dem_features():
+        if feature.GetGeometryRef().Intersects(geometry):
+            return True
+    return False
+
+
+def get_dem_file_paths(geometry: ogr.Geometry) -> list[str]:
+    file_paths = []
+    for feature in get_dem_features():
+        if feature.GetGeometryRef().Intersects(geometry):
+            file_paths.append(feature.GetField('file_path'))
+    return file_paths
+
+
+def utm_from_lon_lat(lon: float, lat: float) -> int:
```

Comment:
This is another good candidate for `hyp3lib.util`, and replaces `saa_func_lib.get_utm_proj`.

These functions all appear unused; we could open a separate PR to remove them:
- `saa_func_lib.get_zome`
- `saa_func_lib.get_utm_proj`
- `saa_func_lib.reproject_gcs_to_utm`